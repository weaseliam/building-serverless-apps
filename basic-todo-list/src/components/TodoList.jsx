import React from 'react';
import PropTypes from 'prop-types';

import TodoItem from './TodoItem';

const TodoList = ({ todos, onToggleTodo }) => (
  <ul>
    {
      todos.map((todo, index) =>
        <TodoItem
          name={todo.name}
          done={todo.done}
          onToggleDone={() => onToggleTodo(todo, index)}
          key={index} />)
    }
  </ul>
);

TodoList.propTypes = {
  todos: PropTypes.array,
  onToggleTodo: PropTypes.func
};

TodoList.defaultProps = {
  todos: [],
  onToggleTodo: () => { }
};

export default TodoList;
