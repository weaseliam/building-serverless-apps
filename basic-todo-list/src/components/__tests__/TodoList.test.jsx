import React from 'react';
import renderer from 'react-test-renderer';

import TodoList from '../TodoList';

test('TodoList renders as expected', () => {
  const component = renderer.create(<TodoList todos={[{ name: "Hello World !" }]} />);

  expect(component.toJSON()).toMatchSnapshot();
});
