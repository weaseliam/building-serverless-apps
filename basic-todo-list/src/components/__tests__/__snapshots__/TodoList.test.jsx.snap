// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders as expected 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    Hello World !
  </li>
</ul>
`;
